File: src/main/java/frc/robot/commands/Drivetrain_Commands/DriveForDistanceCommand.java
package frc.robot.commands.Drivetrain_Commands;

import edu.wpi.first.wpilibj2.command.Command;
import frc.robot.subsystems.DrivetrainFalcon;

class DriveForDistanceCommand extends Command {

    private DrivetrainFalcon drivetrainFalcon;
    private double speed;
    private double distance;

    public DriveForDistanceCommand(DrivetrainFalcon drivetrainFalcon, double speed, double distance) {
        this.drivetrainFalcon = drivetrainFalcon;
        this.speed = speed;
        this.distance = distance;
        addRequirements(drivetrainFalcon);
    }

    @Override
    public void initialize() {
        System.out.println("INFO: DriveForDistanceCommand initialize");
        drivetrainFalcon.arcadeDrive(speed, 0.0, false);
    }

    @Override
    public void execute() {
    }

    @Override
    public boolean isFinished() {
        var averageDist = drivetrainFalcon.getAverageDist();
        return averageDist >= distance;
    }

    @Override
    public void end(boolean interrupted) {
        System.out.println("INFO: DriveForDistanceCommand end");
        drivetrainFalcon.stop();
    }
}

File: src/main/java/frc/robot/commands/Drivetrain_Commands/DriveForDistanceCommand.kt
package frc.robot.commands.Drivetrain_Commands

import edu.wpi.first.wpilibj2.command.CommandBase
import frc.robot.subsystems.DrivetrainFalcon

class DriveForDistanceCommand(private val drivetrainFalcon: DrivetrainFalcon,
                              private val speed: Double,
                              private val distance: Double) : CommandBase() {

    init {
        // each subsystem used by the command must be passed into the addRequirements() method
        addRequirements(drivetrainFalcon)
    }

    override fun initialize() {
        drivetrainFalcon.arcadeDrive(speed, 0.0, squareInputs=false)
        println("INFO: DriveForDistanceCommand initialize")
    }

    override fun execute() {}

    override fun isFinished(): Boolean {
        val averageDist = drivetrainFalcon.getAverageDist()
        if (averageDist >= distance) {
            return true
        }
        return false
    }

    override fun end(interrupted: Boolean) {
        drivetrainFalcon.stop()
        println("INFO: DriveForDistanceCommand end")
    }
}
